File: MediafyDemoAppObjectiveC/MediafyDemoAppObjectiveC/IntegrationCases/AppLovin/AppLovinVideoInterstitialViewController.m
/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */

#import "AppLovinVideoInterstitialViewController.h"

@interface AppLovinVideoInterstitialViewController ()

@property (nonatomic) MAInterstitialAd * interstitialAd;

@end

@implementation AppLovinVideoInterstitialViewController

- (NSString *)adUnitId {
    return @"<AD_UNIT_ID>";
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self loadAd];
}

- (void)loadAd {
    // 1. Create a MAInterstitialAd
    self.interstitialAd = [[MAInterstitialAd alloc] initWithAdUnitIdentifier:self.adUnitId];
    
    // 2. Configure the MAInterstitialAd
    self.interstitialAd.delegate = self;
    
    // 3. Load the ad
    [self.interstitialAd loadAd];
}

// MARK: - MAAdDelegate

- (void)didLoadAd:(MAAd *)ad {
    if (self.interstitialAd != nil && self.interstitialAd.isReady) {
        [self.interstitialAd showAd];
    }
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(MAError *)error {
    NSLog(@"Did fail to load ad with error: %@", error.message);
}

- (void)didFailToDisplayAd:(MAAd *)ad withError:(MAError *)error {
    NSLog(@"Did fail to display ad with error: %@", error.message);
}

- (void)didDisplayAd:(MAAd *)ad {}
- (void)didHideAd:(MAAd *)ad {}
- (void)didClickAd:(MAAd *)ad {}

@end

File: MediafyDemoAppSwift/MediafyDemoAppSwift/IntegrationCases/AppLovin/AppLovinVideoInterstitialViewController.swift
/*
 * Mediafy SDK
 *
 * Copyright 2024 Mars Technologies Ltd. All rights reserved.
 */

import MediafySDK
import AppLovinSDK

fileprivate let adUnitId = "<AD_UNIT_ID>"

final class AppLovinVideoInterstitialViewController: InterstitialBaseViewController {
    
    private var interstitialAd: MAInterstitialAd!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        loadAd()
    }
    
    private func loadAd() {
        // 1. Create a MAInterstitialAd
        interstitialAd = MAInterstitialAd(adUnitIdentifier: adUnitId)
        
        // 2. Configure the MAInterstitialAd
        interstitialAd.delegate = self
        
        // 3. Load the ad
        interstitialAd.load()
    }
}

// MARK: - MAAdDelegate

extension AppLovinVideoInterstitialViewController: MAAdDelegate {
    
    func didLoad(_ ad: MAAd) {
        if interstitialAd != nil && interstitialAd.isReady {
            interstitialAd.show()
        }
    }
   
    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError) {
        print("Did fail to load ad with error: \(error.message)")
    }
    
    func didFail(toDisplay ad: MAAd, withError error: MAError) {
        print("Did fail to display ad with error: \(error.message)")
    }
    
    func didDisplay(_ ad: MAAd) {}
    func didHide(_ ad: MAAd) { }
    func didClick(_ ad: MAAd) {}
}
